src/Scanner.js

```diff
@@ -220,9 +238,7 @@ class DefaultScanner {
             if (this.forms.size < this.options.maxFormsPerPage) {
                 this.forms.set(parentForm, new Form(parentForm, input, this.device, this.matching, this.shouldAutoprompt))
             } else {
-                if (shouldLog()) {
-                    console.log('The page has too many forms, stop adding them.')
-                }
+                this.stopScanner('The page has too many forms, stop adding them.')
```

Comment:
Instead of just not adding fields, we're now actually stopping the scanner. We can revisit and add more complexity later on, but this surely cuts the performance issues here (instead of leaving the `MutationObserver` around doing useless work).